### PR TITLE
LPTimeIntervalView + Objj2 compiler compatible

### PR DESCRIPTION
**Pull request by primalmotion**
# LPTimeInterval

LPTimeInterval view is a widget that allows user to select a date range.

He can select one of the following modes:
- Hours
- Days
- Months
- Year
### Hours mode:

A standard LPCalendar view will be shown, with two steppers to select hours
When he selects something, he will get two dates: 
- start date = selected day at selected start hour
- end date = selected day at selected end hour
### Day mode:

A standard LPCalendar view will be shown
When he selects something, he will get two dates: 
- start date = selected day at 00:00
- end date = selected day at 23:59
### Month mode:

A list of months will be shown
When he selects something, he will get two dates: 
- start date = first day of selected month at 00:00
- end date = last day of selected month at 23:59
### Year mode:

A list of years will be shown
When he selects something, he will get two dates: 
- start date = first day of first month of selected year at 00:00
- end date = last day of last month of selected year at 23:59
# Obj2 compiler compatibility

Fixed imports to works with more strict new compiler

### CODE REVIEW

File: CPDate+FirstLastDate.j [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+*   Author:           Alexandre Wilhelm <alexandre.wilhelm@alcatel-lucent.com>
+*   Description:      CNA Dashboard
+*   Project:          Cloud Network Automation - Nuage - Data Center Service Delivery - IPD
+*
+* Copyright (c) 2011-2012 Alcatel, Alcatel-Lucent, Inc. All Rights Reserved.
+*
+* This source code contains confidential information which is proprietary to Alcatel.
+* No part of its contents may be used, copied, disclosed or conveyed to any party
+* in any manner whatsoever without prior written permission from Alcatel.
+*
```
Comment (aljungberg) [inline CPDate+FirstLastDate.j:8]: This will make it hard to merge the code. :)

File: CPDate+FirstLastDate.j [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    return new Date(aDate.getUTCFullYear(),aDate.getMonth(),1);
+}
+
+/*! Return the first date of an year
+    @param aDate the targeted year
+    @return a new date
+*/
++ (CPDate)firstDateOfYear:(CPDate)aDate
+{
+    return new Date(aDate.getUTCFullYear(),0,1);;
```
Comment (aljungberg) [inline CPDate+FirstLastDate.j]: Not a big deal but the normal code style calls for spaces after commas and no double `;` etc.

**Comment by aljungberg** [comment]
Is there a test app?

File: CPDate+FirstLastDate.j (hunk 1 above)
Comment (Dogild) [inline CPDate+FirstLastDate.j:8]: Really :p ?

**Pull request closed without merging**